protocols/directdefund/directdefund.go

```diff
@@ -97,60 +99,86 @@ func (o Objective) Id() protocols.ObjectiveId {
 	return protocols.ObjectiveId(ObjectivePrefix + o.C.Id.String())
 }
 
-func (o Objective) Approve() Objective {
+func (o Objective) Approve() protocols.Objective {
 	updated := o.clone()
 	// todo: consider case of o.Status == Rejected
 	updated.Status = protocols.Approved
 
-	return updated
+	return &updated
 }
 
-func (o Objective) Reject() Objective {
+func (o Objective) Reject() protocols.Objective {
 	updated := o.clone()
 	updated.Status = protocols.Rejected
-	return updated
+	return &updated
+}
+
+// OwnsChannel returns the channel that the objective is funding.
+func (ddo Objective) OwnsChannel() types.Destination {
+	return ddo.C.Id
+}
+
+// GetStatus returns the status of the objective.
+func (ddo Objective) GetStatus() protocols.ObjectiveStatus {
+	return ddo.Status
+}
+
+func (o Objective) Related() []protocols.Storable {
+	return []protocols.Storable{o.C}
 }
 
 // Update receives an ObjectiveEvent, applies all applicable event data to the DirectDefundingObjective,
 // and returns the updated objective
-func (o Objective) Update(event protocols.ObjectiveEvent) (Objective, error) {
+func (o Objective) Update(event protocols.ObjectiveEvent) (protocols.Objective, error) {
 	if o.Id() != event.ObjectiveId {
-		return o, fmt.Errorf("event and objective Ids do not match: %s and %s respectively", string(event.ObjectiveId), string(o.Id()))
+		return &o, fmt.Errorf("event and objective Ids do not match: %s and %s respectively", string(event.ObjectiveId), string(o.Id()))
 	}
 
 	if len(event.SignedStates) > 0 {
 		for _, ss := range event.SignedStates {
 			if !ss.State().IsFinal {
-				return o, errors.New("direct defund objective can only be updated with final states")
+				return &o, errors.New("direct defund objective can only be updated with final states")
 			}
 			if o.finalTurnNum != ss.State().TurnNum {
-				return o, fmt.Errorf("expected state with turn number %d, received turn number %d", o.finalTurnNum, ss.State().TurnNum)
+				return &o, fmt.Errorf("expected state with turn number %d, received turn number %d", o.finalTurnNum, ss.State().TurnNum)
 			}
 		}
 	}
 
 	updated := o.clone()
 	updated.C.AddSignedStates(event.SignedStates)
-	if event.Holdings != nil {
-		updated.C.OnChainFunding = event.Holdings
+
+	return &updated, nil
+}
+
+func (o Objective) UpdateWithChainEvent(event chainservice.Event) (protocols.Objective, error) {
+	updated := o.clone()
+	de, ok := event.(chainservice.AllocationUpdatedEvent)
+	if !ok {
+		return &updated, fmt.Errorf("objective %+v cannot handle event %+v", updated, event)
+	}
+	// todo: check block number
```

Comment:
can this be included now?

Reply:
This PR keeps the existing logic for reorgs in place. I added this `todo` as I noticed that the direct defunder did not address reorgs. As per above comment, this is important work that should be done at a later point.